scheduler/src/cook/kubernetes/api.clj

```diff
@@ -686,20 +696,25 @@
     :or {pod-priority-class cook-job-pod-priority-class
          pod-supports-cook-init? true
          pod-supports-cook-sidecar? true}}]
-  (let [{:keys [scalar-requests job]} task-request
+  (let [{:keys [scalar-requests job resources]} task-request
         ;; NOTE: The scheduler's adjust-job-resources-for-pool-fn may modify :resources,
         ;; whereas :scalar-requests always contains the unmodified job resource values.
         {:strs [mem cpus]} scalar-requests
         {:keys [docker volumes]} container
         {:keys [image parameters port-mapping]} docker
         {:keys [environment]} command
+        pool-name (some-> job :job/pool :pool/name)
+        ; gpu count is not stored in scalar-requests
```

Comment:
Is it worth mentioning why this is the case?

Reply:
yeah I expanded the comment